xnmt/encoder.py

```diff
@@ -311,6 +321,24 @@ def transduce(self, embed_sent):
   def initial_state(self):
     return PseudoState(self)
 
+
+class TransformerEncoder(BuilderEncoder, Serializable):
+  yaml_tag = u'!TransformerEncoder'
+
+  def __init__(self, context, input_dim=512, layers=1, hidden_dim=512, dropout=0.5, **kwargs):
```

Comment:
Sorry, actually what I meant regarding dropout would be something like:

def __init__(self, context, input_dim=512, layers=1, hidden_dim=512, dropout=None, **kwargs):
  self.dropout = dropout or context.dropout

That way we will be using the globally configured dropout, unless the TransformerEncoder-specific dropout value is set in the config.